src/firebase/notifications.ts

```diff
@@ -57,12 +57,12 @@ export function* handleNotification(
   notificationState: NotificationReceiveState
 ) {
   ValoraAnalytics.track(AppEvents.push_notification_opened, {
-    id: message.data?.id,
+    id: message.data?.id as string,
```

Comment:
there was a change to the type declarations where every key in `data` can be a string or object, but it should not affect these particular properties `https://github.com/invertase/react-native-firebase/pull/7316/files`

Reply:
i know that these properties should always be strings, so i don't think it's worth adding actual type checks, but stumbling across these assertions always gives me a little pause - maybe add a comment saying that these properties are unaffected in our use case?

Reply:
what if there _is_ an object for some reason?

I've seen some places with typeguards like this, what do you think @kathaypacific ?

```typescript
if (
  typeof message.data?.id !== 'string' ||
  typeof message.data?.type !== 'string' ||
  typeof message.data?.ou !== 'string'
) {
  return // this should never happen
}
```

Reply:
TY for the feedback! i've changed it to something like the above :)